python/ray/autoscaler/cli_logger.py

```diff
@@ -513,6 +513,30 @@ def confirm(self, yes, msg, *args, _abort=False, _default=False, **kwargs):
 
         return res
 
+    def prompt(self, msg: str, *args, **kwargs):
+        """Display a confirmation dialog.
+
+        Args:
+            msg (str): The mesage to display to the user before the prompt.
+        """
+        if self.old_style:
+            return
+
+        complete_str = cf.underlined(msg + ":") + " "
+
+        self._print(complete_str, linefeed=False)
+
+        res = None
+        try:
+            ans = sys.stdin.readline()
+            ans = ans.lower()
+            ans = ans.strip()
+        except KeyboardInterrupt:
+            self.newline()
+            res = default
```

Comment:
`default` is undefined as far as I can see. `confirm` gets it from the `_default` keyword argument.

It might be a good idea to drop defaults completely, by returning `""` both when the user presses "enter" with no input and when the user "^C"s.

Otherwise, we should display the default value somehow, but very long defaults will be a problem.

P.S.
I am actually leaning towards having an *optional* `_default` parameter, so if there is a default with `prompt` we don't forget to print it, but if there is not one we should keep asking until we get user input.

Reply:
I left this as is. I don't feel strongly about this, but in a lot of cases, validating the input will be more complex than just checking if it's an empty string or not (e.g., maybe it needs to be an int or something), so either we would need to pass in a lambda for validation or we should do the validation outside of the `prompt` function. So I left it as is for now.